pandas/core/arrays/string_arrow.py

```diff
@@ -433,3 +443,113 @@ def _str_rstrip(self, to_strip=None):
         else:
             result = pc.utf8_rtrim(self._pa_array, characters=to_strip)
         return type(self)(result)
+
+
+class ArrowStringArrayNumpySemantics(ArrowStringArray):
+    # _result_converter = lambda _, result: result.to_numpy(na_value=np.nan)
+    _storage = "pyarrow_numpy"
+
+    @staticmethod
+    def _result_converter(values, na=None):
+        if not isna(na):
+            values = values.fill_null(bool(na))
+        return ArrowExtensionArray(values).to_numpy(na_value=np.nan)
```

Comment:
Why only `if not isna(na)`? I assume we will generally always have to fill nulls? (otherwise the `to_numpy` with nan gives object dtype)

For example (using this branch):

```
n [19]: pd.Series(["a", "b", None], dtype="string[pyarrow]").str.startswith("a")
Out[19]: 
0     True
1    False
2     <NA>
dtype: boolean

In [20]: pd.Series(["a", "b", None], dtype="string[pyarrow_numpy]").str.startswith("a")
Out[20]: 
0     True
1    False
2      NaN
dtype: object
```

If we want that the second example has a proper numpy bool dtype that can be used for boolean indexing, we probably should convert the NaN into False? (either with the `fill_null` call or with `to_numpy(na_value=False)`)

Reply:
Not sure about that. This would break behaviour compared to object dtype, which is something I tried to avoid (and will break a huge number of tests). I think it makes more sense to keep as is for consistency sake

Reply:
Ah, OK, so also the current object-dtype string methods propagate the NaN, wasn't aware of that:

```
In [5]: pd.Series(["a", "b", None], dtype="object").str.startswith("a")
Out[5]: 
0     True
1    False
2     None
dtype: object
```

Personally, I think that's something we should change, but that's for another issue then.

Reply:
> Yeah happy to discuss that, but maybe better as a breaking change for 3.0? Don't know though. Let's open an issue about that

Opened https://github.com/pandas-dev/pandas/issues/54805 for the question whether string predicate methods like startswith should propagate NaN or not